specifyweb/frontend/js_src/lib/components/DataModel/schemaExtras.ts

```diff
@@ -121,6 +122,8 @@ export const schemaExtras: {
       catalognumber.getFormat = (): string | undefined =>
         schema.catalogNumFormatName ||
         LiteralField.prototype.getFormat.call(catalognumber);
+      catalognumber.getUiFormatter = (): UiFormatter | undefined =>
```

Comment:
hmm, this doesn't seem necessary

The default implementation of getUiFormatter is:
```js
  // Returns the UIFormatter for the field specified in the schema config.
  public getUiFormatter(): UiFormatter | undefined {
    return getUiFormatters()[this.getFormat() ?? ''];
  }
```

Which is identical to your implementation.
Is `this.getFormat()` referring to a different function